File: Back/.Net/Modules/Users/BetFriends.Users.Infrastructure/Outbox/InMemoryOutboxAccessor.cs
using BetFriends.Shared.Infrastructure.Outboxes;

namespace BetFriends.Users.Infrastructure.Outboxes;

internal class InMemoryOutboxAccessor : IOutbox
{
    private readonly List<Outbox> outboxes = [];

    public Task AddAsync(Outbox outbox)
    {
        outboxes.Add(outbox);
        return Task.CompletedTask;
    }

    public Task<IEnumerable<Outbox>> GetAllAsync()
        => Task.FromResult(outboxes.Where(x => x.ProcessedOn is null));

    public Task SaveAsync(Outbox outbox)
    {
        return Task.CompletedTask;
    }
}

File: Back/NodeJS/modules/users/src/infrastructure/outbox/InMemoryOutboxAccessor.ts
import { IOutboxAccessor } from "../../../../shared/infrastructure/outbox/IOutboxAccessor";
import { Outbox } from "../../../../shared/infrastructure/outbox/Outbox";

export class InMemoryOutboxAccessor implements IOutboxAccessor {
    private readonly outboxes: Outbox[] = [];
    GetAll(): Promise<Outbox[]> {
        return Promise.resolve(this.outboxes.filter(x => x.HandledAt == undefined));
    }
    Save(outbox: Outbox): Promise<void> {
        this.outboxes.push(outbox);
        return Promise.resolve();
    }

}